src/main/java/org/interledger/cryptoconditions/PreimageSha256Condition.java

```diff
@@ -16,12 +16,16 @@
 
   /**
    * Required-args Constructor.  Constructs an instance of {@link PreimageSha256Condition} based
-   * on a supplied preimage. This constructor variant is intended to be used by developers
-   * wishing to construct a Preimage condition from a secret preimage.
+   * on a supplied preimage.
+   *
+   * Note this constructor is package-private because it is only used from within {@link PreimageSha256Fulfillment}.
+   *
+   * Developers that wish to create a new {@link PreimageSha256Condition} from the preimage should instead create a new
+   * {@link PreimageSha256Fulfillment} and call {@link PreimageSha256Fulfillment#getCondition()}.
    *
    * @param preimage An instance of byte array containing preimage data.
    */
-  public PreimageSha256Condition(final byte[] preimage) {
```

Comment:
This particular change is probably a breaking change. However, we should revisit the Constructor priveleges in all Conditions, so what if we leave this `public` for now, and bump to 0.4.0 when we inspect the other condition classes?

Reply:
I created #74 to track this - if you revert this single change, I think this will be good to commit.